### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Gh-Reza**
## Changes:

- Created the Classroom class
- Create the has-many/belongs-to relationship between Classroom and Student
- Created the Book and the Rental classes
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: classes/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
```
Comment (julie-ify) [inline classes/classroom.rb]: - [ ] Kindly add `students` to this getter and setter.

File: classes/student.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ class Student < Person
   def initialize(age, classroom, name: 'unknown', parent_permession: true)
     super(age, name, parent_permession)
     @classroom = classroom
+    @classroom.add_student(self)
```
Comment (julie-ify) [inline classes/student.rb]: - [ ] The association should have its method. Kindly remove the `@classroom.add_...` part and add the method for the association.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: classes/person.rb [hunk 3]
```diff
@@ -10,6 +10,12 @@ def initialize(age, name: 'unkown', parent_permession: true)
     @name = name
     @parent_permession = parent_permession
     @age = age
+    @rentals = []
```
Comment (julie-ify) [inline classes/person.rb:13]: - [ ] Kindly add `rentals` to the list of `attr_accessor` on top of the class.

File: classes/rental.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :data, :book, :person
+
+  def initialize(data, book, person)
+    @data = data
+    @book = Book
```
Comment (julie-ify) [inline classes/rental.rb]: - [ ] This should be `book`, not `Book`. Kindly fix the error

File: classes/book.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
```
Comment (julie-ify) [inline classes/book.rb]: - [ ] Kindly add `rentals` here

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Gh-Reza,

Good job so far!

## Highlights!
- Good code organization and associations ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — COMMENTED** [review]
Hi @Gh-Reza,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* You likely made a mistake in your submission. This is the associations project but you submitted it using the add basic UI.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Gambit142 — APPROVED** [review]
Hi @Gh-Reza 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Setup linters correctly ✔️ 
- Created new classes based on requirements ✔️ 
- Setup association between classes correctly ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**